src/rgw/rgw_sync_module_es.cc

```diff
@@ -137,21 +113,35 @@ struct ESInfo {
   ESVersion version;
 
   void decode_json(JSONObj *obj);
+
+  std::string get_version_str(){
+    return to_string(version.first) + "." + to_string(version.second);
+  }
+};
+
+struct es_version_decoder {
+  // simple wrapper structure to wrap the es version nested type
+  std::string s;
+  void decode_json(JSONObj *obj) {
+    JSONDecoder::decode_json("number",s,obj);
+  }
+
+  ESVersion get_version() {
+      int _major,_minor;
+      sscanf(s.c_str(), "%d.%d", &_major, &_minor);
```

Comment:
Is there any possibility of parse error (including empty input) here? Remember to check the result of sscanf(), if you'd like to use scanf() here and there's any possibility of error. Or, consider using std::stoi(), which also has error detection (note that the older C atoi() doesn't give you a way to do this) and supports std::string directly.

Reply:
@theanalyst should at least dump error and set defaults to _major and _minor.